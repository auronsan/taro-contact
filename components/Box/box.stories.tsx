import React from 'react';
import { Box } from './index';

export default {
  title: 'components/Box',
  component: Box,
};

export const WithChildren = () => <Box>Content</Box>;

export const WithCustomClassName = () => (
  <Box className="custom-class" id="first-child">
    Content
  </Box>
);

export const WithInlineStyles = () => (
  <Box style={{ backgroundColor: 'red' }} id="first-child">
    Content
  </Box>
);

export const AsADifferentComponent = () => (
  <Box component="span" id="first-child">
    Content
  </Box>
);

export const WithMarginAndPadding = () => (
  <Box m={10} p={20} id="first-child">
    Content
  </Box>
);
