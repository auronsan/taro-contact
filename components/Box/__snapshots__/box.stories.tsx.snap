// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Box AsADifferentComponent smoke-test 1`] = `
<main class="roboto-normal">
  <div style="display: flex; justify-content: center; align-items: center; height: 100vh; flex-direction: column;">
    <span id="first-child"
          data-testid="first-child"
    >
      Content
    </span>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`components/Box WithChildren smoke-test 1`] = `
<main class="roboto-normal">
  <div style="display: flex; justify-content: center; align-items: center; height: 100vh; flex-direction: column;">
    <div>
      Content
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`components/Box WithCustomClassName smoke-test 1`] = `
<main class="roboto-normal">
  <div style="display: flex; justify-content: center; align-items: center; height: 100vh; flex-direction: column;">
    <div class="custom-class"
         id="first-child"
         data-testid="first-child"
    >
      Content
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`components/Box WithInlineStyles smoke-test 1`] = `
<main class="roboto-normal">
  <div style="display: flex; justify-content: center; align-items: center; height: 100vh; flex-direction: column;">
    <div id="first-child"
         data-testid="first-child"
         style="background-color: red;"
    >
      Content
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`components/Box WithMarginAndPadding smoke-test 1`] = `
<main class="roboto-normal">
  <div style="display: flex; justify-content: center; align-items: center; height: 100vh; flex-direction: column;">
    <div id="first-child"
         data-testid="first-child"
         style="margin: 10px; padding: 20px;"
    >
      Content
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;
